fv3core/utils/corners.py

```diff
@@ -532,130 +532,187 @@ def copy_corners_y_stencil(q: FloatField):
             q = q[-3, -2, 0]
 
 
-@gtstencil
-def fill_corners_bgrid_x(q: FloatField):
+class FillCornersBGrid:
+    """
+    Helper-class to fill corners corresponding to the fortran function
+    fill_corners with BGRID=.true. and either FILL=YDir or FILL=YDIR
+    """
+
+    def __init__(
+        self, direction: str, temporary_field=None, origin=None, domain=None
+    ) -> None:
+        self.grid = spec.grid
+        """The grid for this stencil"""
+        if origin is None:
+            origin = self.grid.full_origin()
+        """The origin for the corner computation"""
+        if domain is None:
+            domain = self.grid.domain_shape_full()
+        """The full domain required to do corner computation everywhere"""
+
+        if temporary_field is not None:
+            self._corner_tmp = temporary_field
+        else:
+            self._corner_tmp = utils.make_storage_from_shape(
+                self.grid.domain_shape_full(add=(1, 1, 1)), origin=origin
+            )
+
+        self._copy_full_domain = FrozenStencil(
+            func=copy_defn,
+            origin=origin,
+            domain=domain,
+        )
+
+        """Stencil Wrapper to do the copy of the input field to the temporary field"""
+
+        ax_offsets = axis_offsets(self.grid, origin, domain)
+
+        if direction == "x":
+            self._fill_corners_bgrid = FrozenStencil(
+                func=fill_corners_bgrid_x_defn,
+                origin=origin,
+                domain=domain,
+                externals=ax_offsets,
+            )
+        elif direction == "y":
+            self._fill_corners_bgrid = FrozenStencil(
+                func=fill_corners_bgrid_y_defn,
+                origin=origin,
+                domain=domain,
+                externals=ax_offsets,
+            )
+
+        else:
+            raise NotImplementedError()
+
+    def __call__(self, field: FloatField):
+        self._copy_full_domain(field, self._corner_tmp)
+        self._fill_corners_bgrid(q_in=self._corner_tmp, q_out=field)
```

Comment:
run_on_daint crashes if we don't use keyword args here. It is confusing because in the call method the step before the one that crashes turns the args into keyword args. I must be doing something subtly different here than CopyCorners